src/host_interface/virtio_netdev.c

```diff
@@ -518,6 +519,10 @@ int netdev_init(sgxlkl_config_t* config)
     void* netdev_vq_mem = NULL;
     struct virtio_net_dev* net_dev = NULL;
     char mac[6] = {0xCA, 0xFE, 0x00, 0x00, 0x00, 0x01};
+    srand(time(NULL));
+    for (int i = 3; i < 6; i++) {
+        mac[i] = rand() % 256;
+    }
```

Comment:
`rand()` is a deterministic random number generator and `srand(time(NULL))` is the example that most crypto courses give you as how not to seed it.  An AKS cluster spinning up a bunch of enclaves at the same time and connecting them to a bridge has a very high chance of collisions.  

```suggestion
    // Generate a completely random MAC address
    for (size_t b = 0 ; b < sizeof(mac) ; b += getrandom(&mac[b], sizeof(mac)-b, GRND_RANDOM) {}
    // Set the locally administered bit:
    mac[0] |= 2;
    // Clear the multicast bit (give a unicast MAC address)
    mac[0] &= 0xfe;
```

Reply:
@davidchisnall `getrandom()` needs access to the `/dev/random` device, which isn't available in a docker container.

Reply:
@Francis-Liu Are you sure `/dev/random` is not available in a Docker container? That would break a lot of software, wouldn't it?

Reply:
From the `getrandom(2)` man page:

> Unlike  /dev/random  and  /dev/urandom, getrandom() does not involve the use of pathnames or file descriptors.  Thus, getrandom() can be useful in cases where chroot(2) makes /dev pathnames invisible, and where an application (e.g., a daemon during start-up) closes a file descriptor for one of these files that was opened by a library.

That said, I agree with @letmaik, if `/dev/random` is not available inside Docker containers, that's a big security hole for most microservices.

Reply:
@letmaik @davidchisnall Thank you, I ran a test and verified `getrandom` actually works in a docker container. Moreover, both `dev/random` and `/dev/urandom` still exist in a non-privileged docker container. So, I did not verify the situation as described by the `getrandom(2)` man page, that is when these device files were invisible. Next, I am going to verify this in Kubernetes.

I am curious how things work if you call `getrandom` within SGX-LKL. What is behind `/dev/random` device file within SGX-LKL? Is that the same device as on the host?

Reply:
No, it's entirely implemented inside the enclave by LKL, with the entropy pool initialised using `rdrand`: https://github.com/lsds/sgx-lkl/blob/c5c6da350e61f4fea8bafc14bb07131a765d92ad/src/lkl/setup.c#L1109

Reply:
Thank you @prp. Does this line of code guarantees that calling `getrandom` within LKL would return hardware-generated random number, and its source of randomization is the hardware on the host. In other words, LKL `getrandom` is a real random number generator.

Reply:
I believe that the implicit source of randomness is going to be the `rdrand` instruction executed inside the enclave. This should be properly security-audited though.